scipy/stats/_distribution_infrastructure.py

```diff
@@ -3261,6 +3262,130 @@ def llf(self, sample, /, *, axis=-1):
+    >>> from scipy import stats
+    >>> LogU = stats.make_distribution(stats.loguniform)
+    >>> X = LogU(a=1.0, b=3.0)
+    >>> np.isclose((X + 0.25).median(), stats.loguniform.ppf(0.5, 1, 3, loc=0.25))
+    np.True_
+    >>> X.plot()
+    >>> sample = X.sample(10000, rng=np.random.default_rng())
+    >>> plt.hist(sample, density=True, bins=30)
+    >>> plt.legend(('pdf', 'histogram'))
+    >>> plt.show()
+
+    """
+    # todo: check genpareto, genextreme, genhalflogistic, kstwo, kappa4, tukeylambda
+    parameters = []
+    names = []
+    support = getattr(dist, '_support', (dist.a, dist.b))
+    for shape_info in dist._shape_info():
+        domain = _RealDomain(endpoints=shape_info.endpoints,
+                             inclusive=shape_info.inclusive)
+        param = _RealParameter(shape_info.name, domain=domain)
+        parameters.append(param)
+        names.append(shape_info.name)
+
+    _x_support = _RealDomain(endpoints=support, inclusive=(True, True))
+    _x_param = _RealParameter('x', domain=_x_support, typical=(-1, 1))
+
+    class CustomDistribution(ContinuousDistribution):
+        _parameterizations = ([_Parameterization(*parameters)] if parameters
+                              else [])
+        _variable = _x_param
+
+    def _sample_formula(self, _, full_shape=(), *, rng=None, **kwargs):
+        return dist._rvs(size=full_shape, random_state=rng, **kwargs)
+
+    def _moment_raw_formula(self, n, **kwargs):
+        return dist._munp(int(n), **kwargs)
+
+    def _moment_raw_formula_1(self, order, **kwargs):
+        if order != 1:
+            return None
+        return dist._stats(**kwargs)[0]
+
+    def _moment_central_formula(self, order, **kwargs):
+        if order != 2:
+            return None
+        return dist._stats(**kwargs)[1]
+
+    def _moment_standard_formula(self, order, **kwargs):
+        if order == 3:
+            if dist._stats_has_moments:
+                kwargs['moments'] = 's'
+            return dist._stats(**kwargs)[int(order - 1)]
+        elif order == 4:
+            if dist._stats_has_moments:
+                kwargs['moments'] = 'k'
+            k = dist._stats(**kwargs)[int(order - 1)]
+            return k if k is None else k + 3
+        else:
+            return None
+
+    methods = {'_logpdf': '_logpdf_formula',
+               '_pdf': '_pdf_formula',
+               '_logcdf': '_logcdf_formula',
+               '_cdf': '_cdf_formula',
+               '_logsf': '_logccdf_formula',
+               '_sf': '_ccdf_formula',
+               '_ppf': '_icdf_formula',
+               '_isf': '_iccdf_formula',
+               '_entropy': '_entropy_formula',
+               '_median': '_median_formula'}
+
+    for old_method, new_method in methods.items():
+        # If method of old distribution overrides generic implementation...
+        method = getattr(dist.__class__, old_method, None)
+        super_method = getattr(stats.rv_continuous, old_method, None)
```

Comment:
This implicitly assumes that the type of `dist` inherits directly from `rv_continuous`. Will this also work for distributions that inherit from e.g. `scipy.stats.norm_gen`?

And do you plan to support ducktyped distributions? 
Because for example there's the [`lmo.distributions.l_poly`](https://github.com/jorenham/Lmo/blob/7efe5d42ef744bda6a4edc00732a8e8938e3d493/lmo/distributions/_nonparametric.py#L73) (nonparametric (T)L-moment-based polynomial quantile-) distribution, which has a variable amount of shape parameters (L-moments actually), and therefore couldn't use `rv_continuous` as its base class. But for the most part its public methods should be compatible with that of `rv_continuous`. 

From a typing perspective it would also help to know which methods and signatures the distribution is expected to have. Because that way, I could use a structural typing approach with `typing.Protocol` to capture the wrapped distribution, and pass it as a generic type argument to the type of the returned `CustomDistribution` instance (which I'll mark it with `@type_check_only`).

Reply:
> Will this also work for distributions that inherit from e.g. scipy.stats.norm_gen?

I'm not sure offhand. You're welcome to try. 

> And do you plan to support ducktyped distributions?

Yes. From the top: "Similarly, we can define an interface that allows users to provide custom distribution classes. That way, they don't need to delve into the private methods of ContinuousDistribution to create their own distributions."